lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -318,6 +319,7 @@ def self.resolve_automation_object(uri, user_obj, attr = nil, options = {}, read
     raise "User object not passed in" unless user_obj.kind_of?(User)
     uri = create_automation_object(uri, attr, options) if attr
     options[:uri] = uri
+    User.current_user = user_obj
```

Comment:
@kbrock 
We can but we just need to do it once at the beginning, if we put it in instantiate it will get called multiple times. The workspaces instantiate new workspace at runtime based on the relationships